packages/edit-site/src/components/global-styles/font-sizes/font-size.js

```diff
@@ -0,0 +1,248 @@
+
+	// Get the font size by slug.
+	const fontSize = sizes.find( ( size ) => size.slug === slug );
+
+	// Whether fluid is true or an object, set it to true, otherwise false.
+	const isFluid = !! fontSize.fluid ?? false;
+
+	// Whether custom fluid values are used.
+	const isCustomFluid = typeof fontSize.fluid === 'object';
+
+	const handleNameChange = ( value ) => {
+		updateFontSize( 'name', value );
+	};
+
+	const handleFontSizeChange = ( value ) => {
+		updateFontSize( 'size', value );
+	};
+
+	const handleFluidChange = ( value ) => {
+		updateFontSize( 'fluid', value );
+	};
+
+	const handleCustomFluidValues = ( value ) => {
+		if ( value ) {
+			// If custom values are used, init the values with the current ones.
+			updateFontSize( 'fluid', {
+				min: fontSize.size,
+				max: fontSize.size,
+			} );
+		} else {
+			// If custom fluid values are disabled, set fluid to true.
+			updateFontSize( 'fluid', true );
+		}
+	};
+
+	const handleMinChange = ( value ) => {
+		updateFontSize( 'fluid', { ...fontSize.fluid, min: value } );
+	};
+
+	const handleMaxChange = ( value ) => {
+		updateFontSize( 'fluid', { ...fontSize.fluid, max: value } );
+	};
+
+	const updateFontSize = ( key, value ) => {
+		const newFontSizes = sizes.map( ( size ) => {
+			if ( size.slug === slug ) {
+				return { ...size, [ key ]: value }; // Create a new object with updated key
+			}
+			return size;
+		} );
+
+		setFontSizes( {
+			...fontSizes,
+			[ origin ]: newFontSizes,
+		} );
+	};
+
+	const handleRemoveFontSize = () => {
+		// Navigate to the font sizes list.
+		goBack();
+
+		const newFontSizes = sizes.filter( ( size ) => size.slug !== slug );
+		setFontSizes( {
+			...fontSizes,
+			[ origin ]: newFontSizes,
+		} );
+	};
+
+	const toggleDeleteConfirm = () => {
+		setIsDeleteConfirmOpen( ! isDeleteConfirmOpen );
+	};
+
+	const toggleRenameDialog = () => {
+		setIsRenameDialogOpen( ! isRenameDialogOpen );
+	};
+
+	return (
+		<>
+			<ConfirmDeleteFontSizeDialog
+				fontSize={ fontSize }
+				isOpen={ isDeleteConfirmOpen }
+				toggleOpen={ toggleDeleteConfirm }
+				handleRemoveFontSize={ handleRemoveFontSize }
+			/>
+
+			{ isRenameDialogOpen && (
+				<RenameFontSizeDialog
+					fontSize={ fontSize }
+					isOpen={ isRenameDialogOpen }
+					toggleOpen={ toggleRenameDialog }
+					handleRename={ handleNameChange }
+				/>
+			) }
+
+			<VStack spacing={ 4 }>
+				<HStack justify="space-between" align="flex-start">
+					<ScreenHeader
+						title={ fontSize.name }
+						description={ sprintf(
+							/* translators: %s: font size preset name. */
+							__( 'Manage the font size %s.' ),
+							fontSize.name
+						) }
+						onBack={ () => goTo( '/typography/font-sizes/' ) }
+					/>
+					<FlexItem>
+						<Spacer
+							marginTop={ 3 }
+							marginBottom={ 0 }
+							paddingX={ 4 }
+						>
+							<DropdownMenu
+								trigger={
+									<Button
+										size="small"
+										icon={ moreVertical }
+										label={ __( 'Menu' ) }
```

Comment:
```suggestion
										label={ __( 'Font size options' ) }
```

Improved accessibility. There might be a more appropriate label.